grafana/grafana
# Loki: Remove`TimeSrv` dependency

Part of https://github.com/grafana/grafana/issues/72631 and the the **last step** 🎉 in removing `TimeSrv` dependency. In this PR I am:

- removing `TimeSrv` depenedency from Loki
- removing `getTimeRange` that was calling `timeSrv.geTimeRange()`
- `getTimeRangeParams` has now mandatory `timeRange: TimeRange` param
- `getDataSamples` has now mandatory `timeRange: TimeRange` param. In places where we are calling `getDataSamples`, we either pass provided `timeRange` (should be most of the times), or use `getDefaultTimeRange` from `@grafana/data`.

> [!TIP] 
> ### How to test
> 1. Run Loki `make devenv sources=loki`
> 2. Ensure that in query builder and query editor, the provided options when using `unwrap` are there. 
> <img width="1497" alt="image" src="https://github.com/grafana/grafana/assets/30407135/0f872417-8eaa-49b0-a3f2-7d49decf8202">
> <img width="1056" alt="image" src="https://github.com/grafana/grafana/assets/30407135/19bd6837-cd5b-4a39-b7cb-c185dac15bb4">
